presentation-api/receiver/receiver.js

```diff
@@ -1,65 +1,62 @@
-(() => {
-  let connectionIdx = 0;
-  let messageIdx = 0;
-  const fruitEmoji = {
-    'grapes':      '\u{1F347}',
-    'watermelon':  '\u{1F349}',
-    'melon':       '\u{1F348}',
-    'tangerine':   '\u{1F34A}',
-    'lemon':       '\u{1F34B}',
-    'banana':      '\u{1F34C}',
-    'pineapple':   '\u{1F34D}',
-    'green apple': '\u{1F35F}',
-    'apple':       '\u{1F34E}',
-    'pear':        '\u{1F350}',
-    'peach':       '\u{1F351}',
-    'cherries':    '\u{1F352}',
-    'strawberry':  '\u{1F353}'
-  };
+let connectionIdx = 0;
+let messageIdx = 0;
 
-  function addConnection(connection) {
-    connection.connectionId = ++connectionIdx;
+if (navigator.presentation.receiver) {
+  navigator.presentation.receiver.connectionList
+  .then(list => {
+    list.connections.map(connection => addConnection(connection));
+    list.addEventListener('connectionavailable', function(event) {
+      addConnection(event.connection);
+    });
+  });
+}
 
-    addMessage('New connection #' + connectionIdx);
+function addConnection(connection) {
+  connection.connectionId = ++connectionIdx;
+  addMessage('New connection #' + connectionIdx);
 
-    connection.onmessage = (message) => {
-      messageIdx++;
-      const messageObj = JSON.parse(message.data);
-      const logString = 'Message ' + messageIdx + ' from connection #' +
-          connection.connectionId + ': ' + messageObj.message;
-      addMessage(logString, messageObj.lang);
-      maybeSetFruit(messageObj.message.toLowerCase());
-      connection.send('Received message ' + messageIdx);
-    };
+  connection.addEventListener('message', function(event) {
+    messageIdx++;
+    const data = JSON.parse(event.data);
+    const logString = 'Message ' + messageIdx + ' from connection #' +
+        connection.connectionId + ': ' + data.message;
+    addMessage(logString, data.lang);
+    const fruit = data.message.toLowerCase();
```

Comment:
Might help to factor out a maybeSetFruit(data.message) function here.

Reply:
Done.